business/mesh.go

```diff
@@ -581,22 +582,32 @@ func (in *MeshService) IstiodResourceThresholds() (*models.IstiodThresholds, err
 	conf := config.Get()
 
 	var istioDeployment *v1.Deployment
-	var istioDeploymentConfig = conf.ExternalServices.Istio.IstiodDeploymentName
+	istioDeploymentConfig := conf.ExternalServices.Istio.IstiodDeploymentName
 	var err error
 
 	if IsNamespaceCached(conf.IstioNamespace) {
 		istioDeployment, err = kialiCache.GetDeployment(conf.IstioNamespace, istioDeploymentConfig)
 	} else {
 		istioDeployment, err = in.k8s.GetDeployment(conf.IstioNamespace, istioDeploymentConfig)
 	}
-	if err != nil && !errors.IsNotFound(err) {
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Debugf("Istiod deployment [%s] not found in namespace [%s]", istioDeploymentConfig, conf.IstioNamespace)
+		}
 		return nil, err
 	}
 
-	memoryLimit := istioDeployment.Spec.Template.Spec.Containers[0].Resources.Limits.Memory().AsApproximateFloat64() / 1000000 // in Mb
-	cpuLimit := istioDeployment.Spec.Template.Spec.Containers[0].Resources.Limits.Cpu().AsApproximateFloat64()
-
-	thresholds := models.IstiodThresholds{Memory: memoryLimit, CPU: cpuLimit}
+	thresholds := models.IstiodThresholds{}
+	deploymentContainers := istioDeployment.Spec.Template.Spec.Containers
+	// Assuming that the first container is the istiod container.
+	if len(deploymentContainers) > 0 {
+		if memoryLimit := deploymentContainers[0].Resources.Limits.Memory(); memoryLimit != nil {
+			thresholds.Memory = float64(memoryLimit.ScaledValue(resource.Mega))
+		}
+		if cpuLimit := deploymentContainers[0].Resources.Limits.Cpu(); cpuLimit != nil {
+			thresholds.CPU = cpuLimit.AsApproximateFloat64()
+		}
+	}
```

Comment:
Neither `Resources` nor `Limits` are pointers so if they are empty you get the default empty value rather than a nil pointer:

https://github.com/kubernetes/api/blob/7e99a1ef2ccdd2589e9a41f5083a95c375ada0a2/core/v1/types.go#L2355-L2359
https://github.com/kubernetes/api/blob/7e99a1ef2ccdd2589e9a41f5083a95c375ada0a2/core/v1/types.go#L2271-L2275

But adding tests for this is very easy.